src/main/java/com/google/devtools/build/lib/cmdline/RepositoryMapping.java

```diff
@@ -35,7 +40,10 @@ public abstract class RepositoryMapping {
   // Always fallback to the requested name
   public static final RepositoryMapping ALWAYS_FALLBACK = createAllowingFallback(ImmutableMap.of());
 
-  abstract ImmutableMap<String, RepositoryName> repositoryMapping();
+  // A given String maps to exactly one RepositoryName, but a RepositoryName may have multiple
+  // Strings (that is, local repository names), mapping to it. In other words, the entries would
+  // form a valid ImmutableMap, but not necessarily a valid ImmutableBiMap.
+  abstract ImmutableListMultimap<String, RepositoryName> repositoryMapping();
```

Comment:
I don't think we should make this an `ImmutableListMultimap` just so that we can call `inverse` on it. Maybe have an `ImmutableListMultimap<...> inverse()` method on `RepositoryMapping` instead, or maybe even just `String inverseLookup(RepositoryMapping)`

Reply:
If we don't make it an `Immutable*Multimap`, we would have to iterate over all mapping entries to find the preimage. If you think that that's not going to be costly in terms of CPU time, then I can certainly do it to save memory (and complexity).

Reply:
Pushed that change in a separate commit.

Reply:
I don't believe the inverse is calculated enough to worry about CPU time (since it's only ever done to produce query output). In any case, it certainly isn't enough to justify the API change, IMO.